beetsplug/acousticbrainz.py

```diff
@@ -42,127 +118,102 @@ def commands(self):
-
-            # Get each field and assign it on the item.
-            item.danceable = get_value(
-                "highlevel", "danceability", "all", "danceable",
-            )
-            item.gender = get_value(
-                "highlevel", "gender", "value",
-            )
-            item.genre_rosamerica = get_value(
-                "highlevel", "genre_rosamerica", "value"
-            )
-            item.mood_acoustic = get_value(
-                "highlevel", "mood_acoustic", "all", "acoustic"
-            )
-            item.mood_aggressive = get_value(
-                "highlevel", "mood_aggressive", "all", "aggressive"
-            )
-            item.mood_electronic = get_value(
-                "highlevel", "mood_electronic", "all", "electronic"
-            )
-            item.mood_happy = get_value(
-                "highlevel", "mood_happy", "all", "happy"
-            )
-            item.mood_party = get_value(
-                "highlevel", "mood_party", "all", "party"
-            )
-            item.mood_relaxed = get_value(
-                "highlevel", "mood_relaxed", "all", "relaxed"
-            )
-            item.mood_sad = get_value(
-                "highlevel", "mood_sad", "all", "sad"
-            )
-            item.rhythm = get_value(
-                "highlevel", "ismir04_rhythm", "value"
-            )
-            item.tonal = get_value(
-                "highlevel", "tonal_atonal", "all", "tonal"
-            )
-            item.voice_instrumental = get_value(
-                "highlevel", "voice_instrumental", "value"
-            )
-            item.average_loudness = get_value(
-                "lowlevel", "average_loudness"
-            )
-            item.bpm = get_value(
-                "rhythm", "bpm"
-            )
-            item.chords_changes_rate = get_value(
-                "tonal", "chords_changes_rate"
-            )
-            item.chords_key = get_value(
-                "tonal", "chords_key"
-            )
-            item.chords_number_rate = get_value(
-                "tonal", "chords_number_rate"
-            )
-            item.chords_scale = get_value(
-                "tonal", "chords_scale"
-            )
-            item.initial_key = '{} {}'.format(
-                get_value("tonal", "key_key"),
-                get_value("tonal", "key_scale")
-            )
-            item.key_strength = get_value(
-                "tonal", "key_strength"
-            )
-
-            # Store the data.
-            item.store()
-            if write:
-                item.try_write()
-
-
-def generate_url(mbid, level):
+                self._log.debug(u'Invalid Response: {}', res.text)
+                return {}
+
+        return data
+
+    def _fetch_info(self, items, write):
+        """Get data from AcousticBrainz for the items.
+        """
+        for item in (item for item in items if item.mb_trackid):
```

Comment:
For some reason, this strikes me as slightly less idiomatic than:

```py
for item in items:
    if not item.mb_trackid:
        continue
```

but I definitely don't have a strong argument against it.